assets/blocks/reader-registration/edit.js

```diff
@@ -12,30 +21,205 @@ import './editor.scss';
+									}
+								/>
+								{ lists.length < 1 && (
+									<div style={ { marginBottom: '1.5rem' } }>
+										<Notice isDismissible={ false } status="error">
+											{ __( 'You must select at least one list.', 'newspack-newsletters' ) }
+										</Notice>
+									</div>
+								) }
+								{ Object.keys( listConfig ).length > 0 && <p>{ __( 'Lists', 'newspack' ) }:</p> }
+								{ Object.keys( listConfig ).map( listId => (
+									<ToggleControl
+										key={ listId }
+										label={ listConfig[ listId ].title }
+										checked={ lists.includes( listId ) }
+										disabled={ inFlight }
+										onChange={ () => {
+											if ( ! lists.includes( listId ) ) {
+												setAttributes( { lists: lists.concat( listId ) } );
+											} else {
+												setAttributes( { lists: lists.filter( id => id !== listId ) } );
+											}
+										} }
+									/>
+								) ) }
+								<p>
+									<a href={ newspack_blocks.newsletters_url }>
+										{ __( 'Manage your subscription lists', 'newspack-newsletters' ) }
+									</a>
+								</p>
+							</>
+						) }
+					</PanelBody>
+				) }
 			</InspectorControls>
 			<div { ...blockProps }>
-				<div className="newspack-registration">
+				<div className={ `newspack-registration ${ className }` }>
 					<form onSubmit={ ev => ev.preventDefault() }>
-						<input type="email" placeholder={ placeholder } />
-						<input type="submit" value={ label } />
+						<div className="newspack-registration__form-content">
+							{ newsletterSubscription && lists.length ? (
+								<div className="newspack-registration__lists">
+									{ lists?.length > 1 && (
+										<RichText
+											onChange={ value => setAttributes( { newsletterTitle: value } ) }
+											placeholder={ __( 'Newsletters title…', 'newspack' ) }
+											value={ newsletterTitle }
+											tagName="h3"
+										/>
+									) }
+									<ul>
+										{ lists.map( listId => (
+											<li key={ listId }>
+												<span className="newspack-registration__lists__checkbox">
+													<input type="checkbox" checked readOnly />
+												</span>
+												<span className="newspack-registration__lists__details">
+													<span className="newspack-registration__lists__label">
+														<span className="newspack-registration__lists__title">
+															{ lists.length === 1 ? (
+																<RichText
+																	onChange={ value => setAttributes( { newsletterLabel: value } ) }
+																	placeholder={ __( 'Subscribe to our newsletter', 'newspack' ) }
+																	value={ newsletterLabel }
+																	tagName="span"
+																/>
+															) : (
+																listConfig[ listId ]?.title
+															) }
+														</span>
+														{ displayListDescription && (
+															<span className="newspack-registration__lists__description">
```

Comment:
Now that this is in the WYSIWYG `RichText` editor only, it needs a placeholder in case the value is empty, otherwise it becomes 0px high and uneditable.

Reply:
I'm not sure I follow. The list description (or title) is not editable through `RichText`

Reply:
When only a single list is enabled, the title and description are editable via `RichText`. If the description attribute is empty, the `RichText` field has zero height, so it's not possible to click into it in order to edit the text.

Reply:
The description is never editable through `RichText`. When there's only 1 list, the title of the newsletter is replaced by a `newsletterLabel`, with the placeholder `Subscribe to our newsletter`.

The description remains the one configured and retrieved from the lists, if it doesn't exist it will remain empty.